gulpfile.js

```diff
@@ -106,39 +112,79 @@ gulp.task('translations-feelup', ['prepare-translations'], function () {
                     item.msgstr = [''];
                 }
                 if (0 == item.msgstr[0].length) {
-//                    item.msgid[0] = item.msgid;
+                    // item.msgid[0] = item.msgid;
                     item.msgstr[0] = item.msgid;
                 }
                 return item;
             }
         }))
         .pipe(gulp.dest(languagesFolder));
-});
+}
 
 // Cleanup temporary translation folders.
-gulp.task('cleanup', ['prepare-translations'], function () {
-    var folders = getFolders(languagesFolder);
-
+function cleanup() {
+    prepare_translations();
     return folders.map(function (folder) {
         return gulp.src(path.join(languagesFolder, folder), {read: false})
             .pipe(clean());
     });
-});
+}
 
 // Compile *.po to *.mo binaries for usage.
-gulp.task('compile-translations', ['translations-feelup'], function () {
+function compile_translations() {
+    translations_feelup();
     // Compile POs to MOs.
     return gulp.src(languagesFolder + '*.po')
         .pipe(gettext())
         .pipe(gulp.dest(languagesFolder))
-});
-
-gulp.task('default', [], function () {
-    gulp.run('prepare-source');
-    gulp.run('update-transifex');
-    gulp.run('download-translations');
-    gulp.run('prepare-translations');
-    gulp.run('translations-feelup');
-    gulp.run('cleanup');
-    gulp.run('compile-translations');
-});
\ No newline at end of file
+}
+
+// Run postcss processing for styles.
+function style() {
+    let plugins = [
+        cssnano()
+    ];
+
+    return gulp.src( './assets/scss/**/*.scss' )
+        .pipe( sass() )
+        .pipe( postcss( plugins ) )
+        .pipe( gulp.dest( './assets/css/' ) )
+}
+
+// Compile css only in dev mode.
+function watch() {
+    gulp.watch( './assets/scss/*.scss', style );
+    gulp.watch( './assets/scss/admin/*.scss', style );
+}
+
+// Fully run style and translations compile.
+function build() {
+    style();
+    prepare_source();
+    update_transifex();
+    download_translations();
+    prepare_translations();
+    translations_feelup();
+    cleanup();
+    compile_translations();
+}
+
+exports.prepare_source        = prepare_source;
+exports.update_transifex      = update_transifex;
+exports.download_translations = download_translations;
+exports.prepare_translations  = prepare_translations;
+exports.translations_feelup   = translations_feelup;
+exports.cleanup               = cleanup;
+exports.compile_translations  = compile_translations;
+exports.style                 = style;
+exports.watch                 = watch;
+exports.build                 = build;
+exports.default               = series(
```

Comment:
@bahiirwa I would add the `style` task here too and optimize them, running tasks in parallel which can be run. Like (just an example)

```js
parallel(
  style,
  series(prepare_source, update_transifex, ...)
);
```

Reply:
@bahiirwa Thinking about it again 🤔 I would rather not have transifex in `watch` and `build`, instead would have them in a separate `npm run translation` command.

The reason is, for most day-to-day tasks, we just need the style watcher and maybe in the future a JS uglifier (which we can add when needed). So I propose to 

1. Have `npm run dev` and `npm run build` deal with only style for now.
2. Create a separate command for explicitly extracting translation. This will just create/update the `pot` file.
3. Create a separate command for explicitly updating existing `.po` with the `pot` file.
4. Create a separate command for uploading/syncing translations with transifex.
5. Create a separate command to do the above 3, in series.

Please feel free to modify the tasks/order as you see fit. I am yet to dig deep into exactly how the translations are being uploaded/handled. I will do that once we are done with the changes required in the new gulp tasks.